### PR TITLE
8283719: java/util/logging/CheckZombieLockTest.java failing intermittently

### PR DESCRIPTION
**Pull request by dfuch**
Please find enclosed a patch for 
`8283719: java/util/logging/CheckZombieLockTest.java failing intermittently`

My analysis is that the test fails intermittently because the `FileChannel` created by the test is garbage collected too early, which releases the associated lock before the `FileHandler` is created.
I have replaced the `try { } finally { }` with a `try-with-resource( ) { } finally { }` which will prevent the `FileChannel` from being released before the end of the block. Additional bonus: this should also help with the code that tries to cleanup the files at the end.

Though I haven't been able to reproduce the exact failure yet, I haven't observed the new version of the test failing either.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8283719](https://bugs.openjdk.java.net/browse/JDK-8283719): java/util/logging/CheckZombieLockTest.java failing intermittently

### Reviewers
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8168/head:pull/8168` \
`$ git checkout pull/8168`

Update a local copy of the PR: \
`$ git checkout pull/8168` \
`$ git pull https://git.openjdk.java.net/jdk pull/8168/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8168`

View PR using the GUI difftool: \
`$ git pr show -t 8168`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8168.diff">https://git.openjdk.java.net/jdk/pull/8168.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/util/logging/CheckZombieLockTest.java [hunk 1]
```diff
@@ -241,10 +241,11 @@ private static void testFileHandlerReuse(File writableDir) throws IOException {
             }
 
             if (supportsLocking) {
-                FileChannel fc = FileChannel.open(Paths.get(lock.getAbsolutePath()),
+                handler2 = null;
+                try (FileChannel fc = FileChannel.open(Paths.get(lock.getAbsolutePath()),
                     StandardOpenOption.CREATE_NEW, StandardOpenOption.APPEND,
-                    StandardOpenOption.WRITE);
-                try {
+                    StandardOpenOption.WRITE)) {
```
Comment (AlanBateman) [inline test/jdk/java/util/logging/CheckZombieLockTest.java:247]: Changing this to use try-with-resources looks good. In passing I wonder why it calls lock.getAbsolutePath(), it's okay to call open with a relative path. Also, if you change this test to use import static then you could change this line to:

`try (FileChannel fc = FileChannel.open(Path.of(lock), CREATE_NEW, APPEND, WRITE))`

**Review by AlanBateman — APPROVED** [review]

File: test/jdk/java/util/logging/CheckZombieLockTest.java (hunk 1 above)
Comment (dfuch) [reply test/jdk/java/util/logging/CheckZombieLockTest.java]: Good idea. Done.

**Review by AlanBateman — APPROVED** [review]
Looks fine, you can drop use of getAbsolutePath if you want to, not important here.

**Comment by dfuch** [comment]
> Looks fine, you can drop use of getAbsolutePath if you want to, not important here.

Done - used `File::toPath` since `lock` is a `File`.

**Review by AlanBateman — APPROVED** [review]

**Pull request closed without merging**